privval/signer_validator_endpoint_test.go

```diff
@@ -331,6 +331,9 @@ func TestErrUnexpectedResponse(t *testing.T) {
 			// we do not want to Start() the remote signer here and instead use the connection to
 			// reply with intentionally wrong replies below:
 			rsConn, err := serviceEndpoint.connect()
+			if err != nil {
```

Comment:
`require.NoError` below does the same, no? maybe we should just move `defer` after `require.NotNil`